modules/monitoring-configuring-remote-write-storage.adoc

```diff
@@ -88,27 +116,93 @@ data:
             name: remoteWriteAuth
-            key: user
+            key: <username>
           password:
             name: remoteWriteAuth
-            key: password
+            key: <password>
+----
++
+* bearer token 
++
+A bearer token (`Bearer`) configuration combines a basic authentication username and password with authorization credentials configured to use a specified token (such as an API key).
++
+The following shows a `Bearer` authorization sample that uses a `basicAuth` username and password combined with an authorization token.
+It assumes you have already created a `Bearer` secret with the name `rw-auth` in the `openshift-monitoring` namespace.
+Substitute your username, password, and token for the appropriate `key` values. 
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write.endpoint"
+      - url: "https://basicAuth.remotewrite.com/api/write"
+        basicAuth:
+          username:
+            name: rw-auth
+            key: <username>
+          password:
+            name: rw-auth
+            key: <password>
+      - url: "https://authorization.remotewrite.com/api/write"
+        authorization:
+          type: Bearer
+          credentials:
+            name: rw-auth
+            key: <token_name>
 ----
+* OAuth 2.0
++
+An OAuth 2.0 (`oauth2`) configuration uses the client credentials grant type. Prometheus fetches an access token from the specified remote write endpoint with the specified client access and secret keys.
+You cannot use this method simultaneously with `basic_auth`, `Bearer`, or `sigv2` authentication.
 +
-* The following example configures client TLS authentication:
+The following shows an `oauth2` authentication sample configuration.
+It assumes you have already created an `oauth2` secret with the name `oauth2-credentials` in the `openshift-monitoring` namespace.
+Replace the sample `key`, `scopes`, and `endpointParams` values with your authentication credentials and other values.
 +
 [source,yaml]
 ----
-tlsConfig:
-  ca:
-    <caSecret>
-  cert:
-    <certSecret>
-  keySecret:
-    <keySecret>
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+        - url: https://test.remotewrite.com/api/write
+          remoteTimeout: 30s
+          oauth2:
+            clientId:
+              secret:
+                name: oauth2-credentials
+                key: <id>
+            clientSecret:
+              name: oauth2-credentials
+              key: <secret>
+            tokenUrl: https://example.com/oauth2/token
+            scopes:
+              - <scope_1>
+              - <scope_2>
+            endpointParams:
+              param1: <value_1>
+              param2: <value_2>
```

Comment:
it's better to change the format to below, there are additional spaces
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      remoteWrite:
      - url: https://test.remotewrite.com/api/write
        remoteTimeout: 30s
        oauth2:
          clientId:
            secret:
              name: oauth2-credentials
              key: <id>
          clientSecret:
            name: oauth2-credentials
            key: <secret>
          tokenUrl: https://example.com/oauth2/token
          scopes:
          - <scope_1>
          - <scope_2>
          endpointParams:
            param1: <value_1>
            param2: <value_2>
```

Reply:
fixed

Reply:
see format is still wrong, please see my example